src/cloudai/parser.py

```diff
@@ -49,14 +49,21 @@ def __init__(self, system_config_path: Path) -> None:
         self.system_config_path = system_config_path
 
     def parse(
```

Comment:
IMO this should be inlined into `parse` function:
```py
... # same as it is

plugin_test_path = Path("conf/plugin/test")
try:
    plugin_tests = (
        self.parse_tests(list(plugin_test_path.glob("*.toml")), system) if plugin_test_path.exists() else []
    )
except TestConfigParsingError:
    exit(1)  # exit right away to keep error message readable for users

filtered_tests = tests
test_scenario: Optional[TestScenario] = None
if test_scenario_path:
    test_mapping = {t.name: t for t in tests}
    plugin_test_scenario_mapping = ... # load mapping here
    try:
        test_scenario = self.parse_test_scenario(test_scenario_path, test_mapping, plugin_test_scenario_mapping)
    except TestScenarioParsingError:
        exit(1)  # exit right away to keep error message readable for users
    scenario_tests = set(tr.test.name for tr in test_scenario.test_runs)
    filtered_tests = [t for t in tests if t.name in scenario_tests]

return system, filtered_tests, test_scenario
```

Let's also define two constants for paths:
```py
PLUGINS_ROOT = Path("conf/plugin")
PLUGINS_TESTS_ROOT = PLUGINS_ROOT / "tests"
```

Reply:
Please find the updated code.